### ISSUE TITLE
Tileset editor: Moving a tile on a position which covers the current position.

### ISSUE THREAD

**Issue by Maxs1789** [christopho/solarus-quest-editor#29]
The editor does not allow to move a tile on a position which covers the current position of this tile.

**Comment by Maxs1789** [comment]
A solution is to ignore the current selection in the method [TilesetView::get_items_intersecting_current_area()](https://github.com/christopho/solarus-quest-editor/blob/master/src/gui/tileset_view.cpp#L863).

For example :
```cpp
// Ignore selected items.
for (auto *item : scene->selectedItems()) {
  items.removeAll(item);
}
```
@christopho Do you think that's  a good idea to add a parameter for this case? An `ignore_selection` parameter set to `false` by default.

For now, nothing is broken if we ignore the selection in all cases.

---
The method [TilesetView::get_items_in_current_area()](https://github.com/christopho/solarus-quest-editor/blob/master/src/gui/tileset_view.cpp#L848) isn't used. 

We should remove it or let it and in this case add a same behavior in this method?

**Comment by christopho** [comment]
I think we should ignore the selection in all cases. And document that the function does not return already selected items.
The function TilesetView::get_items_in_current_area() can be removed if it is never used.

**Comment by Maxs1789** [comment]
Commited in master (947b2f5ce4d2c7957183392445f1f01661810b8e)

**Comment by christopho** [comment]
@Maxs1789 Thanks, please also update the changelog next time

**Comment by christopho** [comment]
Merged in 1.4.3
